components/server/src/workspace/workspace-starter.ts

```diff
@@ -1580,6 +1572,22 @@ export class WorkspaceStarter {
                     operations: ["create", "get"],
                 }),
         ];
+
+        if (project?.teamId) {
+            // we don't have an explicit guard for a project,
+            // but we need to ensure that a workspace has access only to the associated project
+            // we assume here that a workspace don't have a way to get other project IDs besides one which we pass
+            // additionally we should be careful not exposing team specific function scopes
```

Comment:
I actually clients should be able to construct such scope, so should we expose team id to the workspace as well? If we want only project id then we have to introduce a project guard.

Reply:
cc @geropl wdyt about it?

Reply:
I don't really understand why we need this scope in the first place. I think it's worth discussing in sync.